R/format-report.R

```diff
@@ -548,9 +548,8 @@ format_traceability_matrix <- function(exports_df, return_vals = FALSE){
     # Unnest tests and testing directories
     exported_func_df <- exports_df %>%
       mutate(
-        test_files = purrr::map_chr(.data$test_files, ~paste(.x, collapse = "\n")),
-        test_dirs = purrr::map_chr(.data$test_dirs, ~paste(.x, collapse = "\n")),
-      )
+        test_files = purrr::map_chr(.data$test_files, ~paste(.x, collapse = "\n"))
+      ) %>% tidyr::unnest(test_dirs)
```

Comment:
I'm not spotting a description of the problem in the PR or in the commit message for bbf34c2 (bug fix: fix display of multiple test directories). If I revert this code change, the tests still pass, so it's not covered there.

Based on testing this, I think this is an unrelated fix that was prompted by you seeing this in the shiny scorecard:

![image](https://github.com/metrumresearchgroup/mpn.scorecard/assets/1297788/e6d95a89-4634-4870-875a-2406fead2617)

With this change, that's now:

![image](https://github.com/metrumresearchgroup/mpn.scorecard/assets/1297788/4d0fa12d-68fc-4aaa-b988-724a21c7282e)


So, the end result looks good to me, but more context would have been appreciated.

Reply:
Yes you are correct that that was the case I ran into. Could add a simple test for this, but dont know if that's necessary

Reply:
FYI I discovered that this change actually introduced a bug. Unnesting removed all rows with `NULL` testing directories:

```r
> exports_df %>%
+       mutate(
+         test_files = purrr::map_chr(.data$test_files, ~paste(.x, collapse = "\n"))
+       ) 
# A tibble: 36 × 5
   exported_function code_file     documentation      test_files            test_dirs
   <chr>             <chr>         <chr>              <chr>                 <list>   
 1 abc.ci            R/bootfuns.q  man/abc.ci.Rd      ""                    <NULL>   
 2 boot              R/bootfuns.q  man/boot.Rd        "parallel-censboot.R" <chr [1]>
 3 boot.array        R/bootfuns.q  man/boot.array.Rd  ""                    <NULL>   
 4 boot.ci           R/bootfuns.q  man/boot.ci.Rd     ""                    <NULL>   
 5 censboot          R/bootfuns.q  man/censboot.Rd    "parallel-censboot.R" <chr [1]>
 6 control           R/bootfuns.q  man/control.Rd     ""                    <NULL>   
 7 corr              R/bootfuns.q  man/corr.Rd        ""                    <NULL>   
 8 cum3              R/bootfuns.q  man/cum3.Rd        ""                    <NULL>   
 9 cv.glm            R/bootfuns.q  man/cv.glm.Rd      ""                    <NULL>   
10 EEF.profile       R/bootpracs.q man/EEF.profile.Rd ""                    <NULL>   
# ℹ 26 more rows
# ℹ Use `print(n = ...)` to see more rows

exports_df %>%
+       mutate(
+         test_files = purrr::map_chr(.data$test_files, ~paste(.x, collapse = "\n"))
+       ) %>% tidyr::unnest(test_dirs)
# A tibble: 2 × 5
  exported_function code_file    documentation   test_files          test_dirs
  <chr>             <chr>        <chr>           <chr>               <chr>    
1 boot              R/bootfuns.q man/boot.Rd     parallel-censboot.R tests    
2 censboot          R/bootfuns.q man/censboot.Rd parallel-censboot.R tests  
```

This is important enough where it could make sense to add a test here